### PR TITLE
Multiple widgets of same relation

### LINKED ISSUE #37675: n:m relations with multiple end-points
**Feature description.**
Consider the following table structure:

image

Linking from trails | pipes | buildings to the cooperative works fine, but in the opposite direction only one relation reference widgets can be used, either to trails, or pipes or buildings.

Here we would like to be able to use the same relation three times but specify different end points each time to trails, pipes or buildings.

Currently, I can place the relation widget three times, but they are kind of "linked", when I change them to trails as end points, all three widgets will have the same endpoint.

cc @m-kuhn

### PR DESCRIPTION
**Pull request by signedav**
The relation editor widget ("cardinality" and "force hide form on add feature") are now configurable in the widget config, means not per relation but per relation editor widget that is created in the drag and drop (like show labels and button visibility). 

relationeditorwidget

This is mainly needed for relations to a reference-table providing multiple many-to-many connections. This implementation resolves #37675

Additionally the dialog opened with double click functionality on field- and relation-items is removed since it had no impact anyway. Same for QML/HTML widget in the "Available Widgets" tree.

### Backward compatibility notes:
On reading a project configured with **older QGIS versions**, the config is taken over and written into the new widget configuration. On changing the settings, the config of the first instance of this relation editor widget is written the old way as well. So on reopening on an older QGIS, the new configuration is readable there. 

On **autogenerated** forms you cannot configure relation editor widgets anymore (since `QgsAttributeRelationEdit` is removed). But if it has been configured in an older QGIS, the attribute form will still read this configuration.

On reading the configuration the old way with `editFormConfig().widgetConfig()` over the **API** or third-party-applications with **meta objects**, this is still possible. On setting it, it will write it into the first instance of the relation editor widget as well.

### The "force hide form popup on add feature" is working weird
When I activate the checkbox on one widget (or on one relation) then it's taken over for all the relation editor widgets. This bug (?) has been there before and is not fixed in this PR. I'll plan a follow up for that.

### CODE REVIEW

File: python/gui/auto_generated/editorwidgets/qgsrelationwidgetwrapper.sip.in [hunk 1]
```diff
@@ -113,6 +113,39 @@ Returns the buttons which are shown
...
+and for the vectorLayerTools (if true).
+This flag will override the layer and general settings regarding the automatic
+opening of the attribute form dialog when digitizing is completed.
+
+.. versionadded:: 3.16
+%End
+
+    QVariant nmRelationId() const;
+%Docstring
+Determines the nmRelationId used for the many-to-many cardinality.
```
Comment (m-kuhn) [inline python/gui/auto_generated/editorwidgets/qgsrelationwidgetwrapper.sip.in:135]:
```suggestion
Determines the relation id of the second relation involved. Used for many-to-many cardinality.
```

@signedav does this help readability?

File: src/core/qgseditformconfig.cpp [hunk 2]
```diff
@@ -96,6 +134,13 @@ bool QgsEditFormConfig::setWidgetConfig( const QString &widgetName, const QVaria
     return false;
   }
 
+  //for legacy use it writes the relation editor configuration into the first instance of the widget
+  if ( config.contains( QStringLiteral( "force-suppress-popup" ) ) || config.contains( QStringLiteral( "nm-rel" ) ) )
+  {
+    QgsMessageLog::logMessage( QStringLiteral( "Relation settings should be done for the specific widget instance like attributeEditorRelation->setNmRelationId() instead." ) );
```
Comment (m-kuhn) [inline src/core/qgseditformconfig.cpp:140]:
```suggestion
    QgsMessageLog::logMessage( QStringLiteral( "Deprecation Warning: Trying to set a relation config directly on the relation %1. Relation settings should be done for the specific widget instance instead. Use attributeEditorRelation->setNmRelationId() or attributeEditorRelation->setForceSuppressFormPopup() instead." ).arg( widgetName ) );
```

**Comment by nyalldawson** [comment]
Why the move of all the classes from gui to core? I didn't look in depth, but couldn't immediately see a good reason for this change.

**Comment by nyalldawson** [comment]
Ignore that, I read the diff wrong!

File: python/gui/auto_generated/editorwidgets/qgsrelationwidgetwrapper.sip.in (hunk 1 above)
Comment (signedav) [reply python/gui/auto_generated/editorwidgets/qgsrelationwidgetwrapper.sip.in:135]: Without the "Used for many-to-many cardinality." part...

**Comment by signedav** [comment]
No stale. @m-kuhn can we finalize that?

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**